src/components/Widget.tsx

```diff
@@ -105,7 +106,15 @@ export type WidgetProps = {
 }
 
 export default function Widget(props: PropsWithChildren<WidgetProps>) {
-  const { children, theme, provider, dialog: userDialog, className, onError } = props
+  return <TestableWidget {...props} initialAtomValues={undefined} />
+}
+
+export interface TestableWidgetProps extends WidgetProps {
+  initialAtomValues?: Iterable<readonly [Atom<unknown>, unknown]>
+}
+
+export function TestableWidget(props: PropsWithChildren<TestableWidgetProps>) {
+  assert(props.initialAtomValues && process.env.NODE_ENV === 'test', 'initialAtomValues may only be used for testing')
```

Comment:
(This is my mistake, but I don't want to push too many commits to your branch in case you want to back it out.)

This fails if unset of if not in `'test'`. Instead, it should be:

    assert(
      props.initialAtomValues && process.env.NODE_ENV === 'test' ?? true,
      'initialAtomValues may only be used for testing'
    )